utils/organizeHistory.js

```diff
@@ -1,19 +1,19 @@
 import createHeightPercentages from "./createHeightPercentages";
 import filterByDate from "./filterByDate";
 
-export default function organizeHistory(lifetimeArr, days) {
+export default function organizeHistory(arr, days) {
```

Comment:
Nice effort working through this logic.

It looks like we are using parallel arrays (also know as associative arrays).
Often a JavaScript object (also known as a hash table) are better suited for grouping related data. Choosing the right data structure can greatly simplify our code.

Reply:
I need to table this. I've spent about an hour and I'm still not sure how I can work it out as a hashtable.

createHeightPercentages needs the fully populated lengths array in order to work properly. This is because createHeightPercentages relies on the maximum of those lengths.

I tried doing a hashtable like below:

 let history = {
    day0: {
               day: [...],
               length: 4
    },
    day1: {
               day: [...],
               length: 5
    },
    ...,
    day6: {
               day: [...],
               length: 8
    },
  };
  
  but this doesn't help translate percentages.
  
  I'd appreciate any additional insight on this topic